### DISCUSSION TITLE
filtering sessions with metadata

### DISCUSSION THREAD

**Discussion by litagent** [langfuse/langfuse#990] [Ideas]
Currently sessions are filtering using on USER id, can we also have metadata filtering also.

**Comment by clemra** [comment]
Thank you @litagent. We'll prioritise a rework of our tables and add more data (e.g. also cost, sorting etc.)

**Reply by litagent** [reply]
thanks @clemra , any ETA for this.

**Reply by clemra** [reply]
Hi @litagent 
just wanted to let you know that we have now started to work on this and will ship soon. Will ping you once it's live.

**Reply by snikch** [reply]
Did this ever end up happening?

**Reply by marcklingen** [reply]
We currently focus on making changes related to v3 (more on this [here](https://github.com/orgs/langfuse/discussions/1902#discussioncomment-9729809)). This will unblock many new filters that we want to add which currently add performance issues in large projects

**Reply by nonlocalStream** [reply]
@marcklingen Would v3 support filtering session by metadata/tags? Can't seem to find the answer the link above

**Comment by elmdecoste** [comment]
We're also really interested in this!

**Comment by cryoff** [comment]
Is it maybe possible now to add the metadata to a Session via python SDK in v3?

**Comment by Dragos-P-Popa** [comment]
This is a much needed addition!

**Comment by danpe** [comment]
Hoping for that feature as well 😁

**Comment by lasryaric** [comment]
SAme here, we need this! 🙏

**Reply by mrPsycox** [reply]
same!

**Comment by marcklingen** [comment]
Tracking this here: https://github.com/langfuse/langfuse/issues/6091
cc @davidmc971
